src/models/Observation.js

```diff
@@ -1,10 +1,101 @@
+import { formatCameraDate } from "../sharedHelpers/dateAndTime";
+import { getUserId } from "../components/LoginSignUp/AuthenticationService";
 
 class Observation {
+  static FIELDS = {
+    captive: true,
+    comments: Comment.COMMENT_FIELDS,
+    created_at: true,
+    description: true,
+    geojson: true,
+    geoprivacy: true,
+    id: true,
+    identifications: Identification.ID_FIELDS,
+    latitude: true,
+    location: true,
+    longitude: true,
+    observation_photos: ObservationPhoto.OBSERVATION_PHOTOS_FIELDS,
+    place_guess: true,
+    quality_grade: true,
+    taxon: Taxon.TAXON_FIELDS,
+    time_observed_at: true,
+    user: User.USER_FIELDS
+  }
+
+  static createNewObservation( obs ) {
+    return {
+      ...obs,
+      captive: false,
+      geoprivacy: "open",
+      owners_identification_from_vision: false,
+      // project_ids: [],
+      uuid: uuid.v4( )
+    };
+  }
+
+  static async createObsWithNoEvidence( ) {
+    const latLng = await fetchUserLocation( );
+    const obs = {
+      ...latLng,
+      observed_on_string: createObservedOnStringForUpload( )
+    };
+    return Observation.createNewObservation( obs );
+  }
+
+  static async createObsWithPhotos( observationPhotos, observedOn ) {
+    const latLng = await fetchUserLocation( );
+    const obs = {
+      ...latLng,
+      observed_on_string: observedOn || createObservedOnStringForUpload( ),
+      observationPhotos
+    };
+    return Observation.createNewObservation( obs );
+  }
+
+  static async createObsWithSounds( ) {
+    const sound = await ObservationSound.createNewSound( );
+    const latLng = await fetchUserLocation( );
+    const obs = {
+      ...latLng,
+      observed_on_string: createObservedOnStringForUpload( ),
+      observationSounds: [sound]
+    };
+    return Observation.createNewObservation( obs );
+  }
+
+  static async createObsPhotos( photos, createPhoto ) {
+    return await Promise.all( photos.map( async photo => {
+      return await createPhoto( photo );
+    } ) );
+  }
+
+  static async createMutipleObsFromGalleryPhotos( obs ) {
+    return Promise.all( obs.map( async ( { photos } ) => {
+      // take the observed_on_string time from the first photo in an observation
+      const observedOn = formatDateAndTime( photos[0].timestamp );
+      const obsPhotos = await Observation.createObsPhotos( photos, ObservationPhoto.formatObsPhotoFromGallery );
+      return await Observation.createObsWithPhotos( obsPhotos, observedOn );
+    } ) );
+  }
+
+  static async createObsFromNormalCamera( photos ) {
+    // take the observed_on_string time from the first photo in an observation
+    const observedOn = formatCameraDate( photos[0].metadata["{Exif}"].DateTimeOriginal );
+    const obsPhotos = await Observation.createObsPhotos( photos, ObservationPhoto.formatObsPhotoFromNormalCamera );
+
+    return await Observation.createObsWithPhotos( obsPhotos, observedOn );
+  }
+
```

Comment:
I'm confused again. I thought the goal was to remove the logic of creating an observation from the NormalCamera and put it in the model instead. Are you saying that the NormalCamera *should* be creating an observation and passing that to ObsEdit? I'm unclear on where you want `const observation = Observation.new( );` to live.

Reply:
K, sorry about that. I think I am over-scoping this PR so let's just ignore this issue for now. I'll just resolve this thread.

That said, here's where I'm coming from: I think it's good to have *reusable* CRUD code in the models, enough so that they provide tools to make it easier to work with that data in numerous contexts, but not so much that you're essentially dropping single-use methods in the model. Something like `createMutipleObsFromGalleryPhotos()` is very, very specific and should probably either not be a method or be defined closer to where it gets used, but something like `new()` that creates a new record with reasonable defaults or the methods you have that map between the Realm schema and the API schema are highly reusable and composable.

So, when I see stuff in, say, `NormalCamera` that's setting attributes like `_updated_at`, that seems like it should live in the model b/c we want to set that attribute *every* time we save a record, but when I see a method like `createMutipleObsFromGalleryPhotos()` in the model I wonder why it's there b/c I can't imagine it will get reused.

As to where a method like `new()` would be defined, that's part of what I was exploring in the `realm-in-models` branch, e.g. in the [Record](https://github.com/inaturalist/iNaturalistReactNative/blob/realm-in-models/src/models/Record.js) class, where there's a `new()` method that all model instances inherit and can override, like [Observation](https://github.com/inaturalist/iNaturalistReactNative/blob/realm-in-models/src/models/Observation.js#L36) does. `Record.new()` literally just makes a new instance with the properties in the arg it receives, but `Observation.new` also does things like set the default geoprivacy.